File: iosApp/Sources/Theme/ShuttlRadius.swift
import CoreGraphics

/// The shape scale, as raw radii.
///
/// Mirrors androidApp's ShuttlRadius.kt number for number. Use with
/// `.clipShape(RoundedRectangle(cornerRadius: ShuttlRadius.large))`, or
/// `Capsule()` where Android would use `pill`.
enum ShuttlRadius {
    static let extraSmall: CGFloat = 8
    static let small: CGFloat = 12
    static let medium: CGFloat = 16
    static let large: CGFloat = 20
    static let extraLarge: CGFloat = 28

    /// Buttons, chips, badges, tab pills. `Capsule()` is equivalent and
    /// preferred in SwiftUI; this exists so the two platforms' scales can be
    /// compared token for token.
    static let pill: CGFloat = 999
}

File: androidApp/src/main/java/com/badmintontracker/android/ui/theme/ShuttlRadius.kt
package com.badmintontracker.android.ui.theme

import androidx.compose.ui.unit.dp

/**
 * The shape scale, as raw radii.
 *
 * [pill] has no slot in M3's [androidx.compose.material3.Shapes], which is why
 * this object exists alongside ShuttlShapes rather than inside it. Mirrors
 * iosApp's ShuttlRadius.swift number for number.
 */
object ShuttlRadius {
    val extraSmall = 8.dp
    val small      = 12.dp
    val medium     = 16.dp
    val large      = 20.dp
    val extraLarge = 28.dp

    /** Buttons, chips, badges, tab pills. Larger than any surface it clips. */
    val pill       = 999.dp
}
